File: 프로그래머스/1/42840. 모의고사/모의고사.py
# 수열 만드는 함수
def make_arr(arr):
    arr = (arr * (cnt // len(arr))) + arr[:cnt % len(arr)]
    return arr


def solution(answers):
    tmp = []
    global cnt
    cnt = len(answers)
    
    first = [1, 2, 3, 4, 5]
    second = [2, 1, 2, 3, 2, 4, 2, 5]
    third = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    
    first = make_arr(first)
    second = make_arr(second)
    third = make_arr(third)
    
    
    # 문제를 채점하는 함수
    def is_correct(arr):
        count = 0;
        for idx in range(cnt):
            if arr[idx] == answers[idx]:
                count += 1
        return count
    
    for people in [first, second, third]:
        tmp.append(is_correct(people))
        
    answer = []
    
    max_num = max(tmp)
    
    for idx, success_cnt in enumerate(tmp):
        if success_cnt == max_num:
            answer.append(idx + 1)
    return answer




    
    
    

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;

class Solution {
    public int[] solution(int[] answers) {
    
        int[] first = {1, 2, 3, 4, 5};
        int[] second = {2, 1, 2, 3, 2, 4, 2, 5};
        int[] third = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
         
        int first_score = 0;
        int second_score = 0;
        int third_score = 0;
        
        for (int i = 0; i < answers.length; i++) {
                if (first[i % first.length] == answers[i]) {
                    first_score += 1;
                }
            
               if (second[i % second.length] == answers[i]) {
                    second_score += 1;
                }
                if (third[i % third.length] == answers[i]) {
                    third_score += 1;
                }
        }
        
        ArrayList<Integer> arr = new ArrayList<>();
        int maxScore = Math.max(first_score, Math.max(second_score, third_score));
        
        if (first_score == maxScore) arr.add(1);
        if (second_score == maxScore) arr.add(2);
        if (third_score == maxScore) arr.add(3);
        
        return arr.stream().mapToInt(Integer::intValue).toArray();
    }
}